### PR TITLE
ServiceAccounts: Fix errors returning API key and simplify conditions

### PR DESCRIPTION
**Pull request by Jguer**
**What this PR does / why we need it**:

- Fix errors returning API key in service accounts
- simplify if conditions

**Which issue(s) this PR fixes**:

Fixes #

**Special notes for your reviewer**:

### CODE REVIEW

**Review by alexanderzobnin — APPROVED** [review]

File: pkg/services/serviceaccounts/api/token.go [hunk 1]
```diff
@@ -45,38 +45,40 @@ func (api *ServiceAccountsAPI) ListTokens(ctx *models.ReqContext) response.Respo
...
-				v := time.Unix(*t.Expires, 0)
-				expiration = &v
-				if !isExpired && (*expiration).Before(time.Now().Add(sevenDaysAhead)) {
-					secondsUntilExpiration = time.Until(*expiration).Seconds()
-				}
-			}
+	saTokens, err := api.store.ListTokens(ctx.Req.Context(), ctx.OrgId, saID)
+	if err != nil {
+		return response.Error(http.StatusInternalServerError, "Internal server error", err)
+
```
Comment (alexanderzobnin) [inline pkg/services/serviceaccounts/api/token.go:51]:
```suggestion
```

File: pkg/services/serviceaccounts/api/api.go [hunk 2]
```diff
@@ -256,11 +256,11 @@ func (api *ServiceAccountsAPI) HideApiKeysTab(ctx *models.ReqContext) response.R
...
 // POST /api/serviceaccounts/migrate
 func (api *ServiceAccountsAPI) MigrateApiKeysToServiceAccounts(ctx *models.ReqContext) response.Response {
-	if err := api.store.MigrateApiKeysToServiceAccounts(ctx.Req.Context(), ctx.OrgId); err == nil {
-		return response.Success("API keys migrated to service accounts")
-	} else {
+	if err := api.store.MigrateApiKeysToServiceAccounts(ctx.Req.Context(), ctx.OrgId); err != nil {
 		return response.Error(http.StatusInternalServerError, "Internal server error", err)
 	}
+
+	return response.Success("API keys migrated to service accounts")
```
Comment (eleijonmarck) [inline pkg/services/serviceaccounts/api/api.go:263]: nit: ```suggestion
	return response.Success("API keys migrated to service accounts with attached service account tokens")
```

File: pkg/services/serviceaccounts/api/api.go [hunk 3]
```diff
@@ -269,11 +269,12 @@ func (api *ServiceAccountsAPI) ConvertToServiceAccount(ctx *models.ReqContext) r
...
 	}
-	if err := api.store.MigrateApiKey(ctx.Req.Context(), ctx.OrgId, keyId); err == nil {
-		return response.Success("Service accounts converted")
-	} else {
+
+	if err := api.store.MigrateApiKey(ctx.Req.Context(), ctx.OrgId, keyId); err != nil {
 		return response.Error(http.StatusInternalServerError, "Error converting API key", err)
 	}
+
+	return response.Success("Service accounts converted")
```
Comment (eleijonmarck) [inline pkg/services/serviceaccounts/api/api.go:277]: naming convention is hard 🤣 

```suggestion
	return response.Success("Service accounts migrated")
```

File: pkg/services/serviceaccounts/api/api.go [hunk 4]
```diff
@@ -282,10 +283,11 @@ func (api *ServiceAccountsAPI) RevertApiKey(ctx *models.ReqContext) response.Res
 	if err != nil {
 		return response.Error(http.StatusBadRequest, "Key ID is invalid", err)
 	}
+
 	if err := api.store.RevertApiKey(ctx.Req.Context(), keyId); err != nil {
-		return response.Error(http.StatusInternalServerError, "Error reverting API key", err)
+		return response.Error(http.StatusInternalServerError, "Error reverting to API key", err)
 	}
-	return response.Success("API key reverted")
+	return response.Success("Reverted to API key")
```
Comment (eleijonmarck) [inline pkg/services/serviceaccounts/api/api.go:290]:
```suggestion
	return response.Success("Reverted service account to API key")
```

**Review by eleijonmarck — APPROVED** [review]
nits, otherwise looks good 👍

**Pull request merged**